pypsa/linopf.py

```diff
@@ -1070,7 +1070,10 @@ def save_optimal_capacities(n, iteration, status):
     s_nom_orig = n.lines.s_nom.copy()
     p_nom_orig = n.links.p_nom.copy()
     n.lines[['s_nom', 's_nom_extendable']] = n.lines['s_nom_opt'], False
-    n.links[['p_nom', 'p_nom_extendable']] = n.links['p_nom_opt'], False
+    for i in n.links.index:
+      if n.links.carrier == "DC":
+        n.links.loc[i,'p_nom'] = n.links.loc[i,'p_nom_opt']
+        n.links.loc[i,'p_nom_extendable'] = False
```

Comment:
Great! I would just comment that such loops are quite inefficient. The better way to do it would be:

```suggestion
dc_links_b = n.links.carrier == "DC"
n.links.loc[dc_links_b, "s_nom"] = n.links.loc[dc_links_b, "s_nom_opt"]
n.links.loc[dc_links_b, "s_nom_extendable"] = False
```
@FabianHofmann I was wondering whether we could also lift the capacity fix of links overall? The reason for fixing was just to prevent AC lines from changing impedances.